### PR TITLE
adds a sample organization set of users and groups (#2963)

### PR DESCRIPTION
**Pull request by punkle**
## Hey, I just made a Pull Request!

The top level is acme-corp. Here is the graph

```
              acme-corp
                  |
              infrastructure
                  |
            _______________
           |               |
   backstage             boxoffice
     |    |              |    |
  team-a team-b      team-c  team-d
```

The members of those groups are as follows:
```
team-a:          team-b:          team-c:           team-d:
Janelle Dawe     Jenny Doe        Frank Tiernan     Eva MacDowell
Nigel Manning    Jonathon Page    Tara MacGovern    Lucy Sheehan
Breanna Davison  Amelia Park      Sarah Gilroy
                 Justine Barrow   Peadar MacMahon
                 Colette Brock    Calum Leavy
```

### CODE REVIEW

**Comment by iain-b** [comment]
Out of curiosity why not model acme-corp as a group with empty ancestors? I know it doesn't really add much but it seems more complete to me.

**Comment by freben** [comment]
Hey! This is great. Maybe a small restructuring since it's a lot of files - 

Would it make sense to make one file per team, named after the team, that has the team itself and its users (as several subdocuments separated by `----`)? And then another file that holds the org structure.

Something like: 

- `examples/acme/team-X.yaml`: contains the group team-X, and the users of team X
- `examples/acme/org.yaml`: contains the groups acme-corp, infrastructure, backstage, and roadie
- `examples/acme-corp.yaml`: points to all of the above

does that make sense?

**Review by stefanalund — CHANGES_REQUESTED** [review]
Please remove the Roadie naming.

**Comment by punkle** [comment]
@iain-b latest commit, adds acme-group with no parent (excluded).

@freben latest commit adds the files into the structure you have suggested.

@stefanalund latest commit, replaces naming with "boxoffice". Let me know if this is ok. :)

**Review by stefanalund — APPROVED** [review]
Thanks! 😄

File: packages/catalog-model/examples/acme/infrastructure-group.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: backstage.io/v1alpha1
+kind: Group
+metadata:
+  name: infrastructure
+  description: The infra business unit
+spec:
+  type: business-unit
```
Comment (freben) [inline packages/catalog-model/examples/acme/infrastructure-group.yaml:7]: both these levels (infrastructure and boxoffice/backstage) have the same type - i don't know enough about orgs to know if there's some other appropriate term for something between a team and a business unit :)

File: packages/catalog-model/examples/acme/team-a-group.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+apiVersion: backstage.io/v1alpha1
+kind: Group
+metadata:
+  name: team-a
+  description: Team A
+spec:
+  type: business-unit
```
Comment (freben) [inline packages/catalog-model/examples/acme/team-a-group.yaml:7]: team?

File: packages/catalog-model/examples/acme/team-b-group.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+apiVersion: backstage.io/v1alpha1
+kind: Group
+metadata:
+  name: team-b
+  description: Team B
+spec:
+  type: business-unit
```
Comment (freben) [inline packages/catalog-model/examples/acme/team-b-group.yaml:7]: team?

File: packages/catalog-model/examples/acme/team-c-group.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+apiVersion: backstage.io/v1alpha1
+kind: Group
+metadata:
+  name: team-c
+  description: Team C
+spec:
+  type: business-unit
```
Comment (freben) [inline packages/catalog-model/examples/acme/team-c-group.yaml:7]: team?

File: packages/catalog-model/examples/acme/team-d-group.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+apiVersion: backstage.io/v1alpha1
+kind: Group
+metadata:
+  name: team-d
+  description: Team D
+spec:
+  type: business-unit
```
Comment (freben) [inline packages/catalog-model/examples/acme/team-d-group.yaml:7]: team?

File: packages/catalog-model/examples/acme/infrastructure-group.yaml (hunk 1 above)
Comment (punkle) [reply packages/catalog-model/examples/acme/infrastructure-group.yaml:7]: Is this field freeform text? How about department?

Comment (freben) [reply packages/catalog-model/examples/acme/infrastructure-group.yaml:7]: Yeah, or rather more like an enum - it is up to the org themselves to decide what terminology they use, we don't enforce one but lowercase-with-dashes is good in any way. Department sounds fine!

**Comment by freben** [comment]
@punkle just that last fix of the department + team, and we're good to go

**Comment by punkle** [comment]
Thanks @freben and apologies for the delay. There was a few groups we didnt discuss. Here is what I went with. Let me know and if this looks ok:

acme-corp: organization
infrastructure: department
backstage: sub-department
box-office: sub-department
team-n: team

File: packages/catalog-model/examples/acme/backstage-group.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+apiVersion: backstage.io/v1alpha1
+kind: Group
+metadata:
+  name: backstage
+  description: The backstage business unit
```
Comment (freben) [inline packages/catalog-model/examples/acme/backstage-group.yaml:5]: the texts still say business unit :) but no big deal

Comment (punkle) [reply packages/catalog-model/examples/acme/backstage-group.yaml]: I fixed this in the next commit.

**Review by freben — APPROVED** [review]

File: packages/catalog-model/examples/acme/infrastructure-group.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+kind: Group
+metadata:
+  name: infrastructure
+  description: The infra department
+spec:
+  type: department
+  parent: acme-corp
+  ancestors: [acme-corp]
+  children: [backstage, boxoffice]
+  descendants: [backstage, boxoffice, team-a, team-b, team-c, team-d]
```
Comment (jcarres-mdsol) [inline packages/catalog-model/examples/acme/infrastructure-group.yaml:11]: This is very not-DRY. If team-b disappears from the hierarchy, now I have to fix multiple files.
Why can't we just do parent and children and create the tree from that?

**Comment by jcarres-mdsol** [comment]
How does this connect to a service? I.e. I'd like to express that this service is maintained by group Team-A

File: packages/catalog-model/examples/acme/infrastructure-group.yaml (hunk 7 above)
Comment (freben) [reply packages/catalog-model/examples/acme/infrastructure-group.yaml:11]: Yep! These fields are about to disappear, as part of building out the [concept of relations](https://github.com/spotify/backstage/issues/1964). Work is undergoing at this very moment on that. You would for example in the end also want to only express `Group.spec.parent`, or only `Group.spec.children` (depending on what direction your data source expresses them as), not both - they form the same tree either way

**Comment by freben** [comment]
@jcarres-mdsol 

```yaml
spec:
  owner: team-a
```

if the owner is a group and in the same namespace. It's a shorthand form for

```yaml
spec:
  owner: group:default/team-a
```

See the docs on [entity references](https://backstage.io/docs/features/software-catalog/references).

**Pull request merged**